# HG changeset patch
# User Haoyu Bai <baihaoyu@gmail.com>
# Date 1274854955 -28800
# Branch T259
# Node ID 6c6f77d2e52547041ee5fea1f650395c6c97535c
# Parent  66224ddfa9c4d53c7f78fb032373fd51d479e3fe
test cases for __future__.division with C ints

diff -r 66224ddfa9c4 -r 6c6f77d2e525 tests/errors/future_division.pyx
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/tests/errors/future_division.pyx	Wed May 26 14:22:35 2010 +0800
@@ -0,0 +1,8 @@
+from __future__ import division
+
+cdef int a, b = 1, c = 2
+a = b / c
+
+_ERRORS = u'''
+4:6: Cannot assign type 'double' to 'int'
+'''
diff -r 66224ddfa9c4 -r 6c6f77d2e525 tests/run/future_division.pyx
--- a/tests/run/future_division.pyx	Wed May 26 01:26:37 2010 +0800
+++ b/tests/run/future_division.pyx	Wed May 26 14:22:35 2010 +0800
@@ -13,6 +13,21 @@
     """
     return x/y, x//y
 
+def cdoit(int x, int y):
+    """
+    >>> cdoit(1,2)
+    (0.5, 0.0)
+    >>> cdoit(4,3)
+    (1.3333333333333333, 1.0)
+    >>> cdoit(4,3.0)
+    (1.3333333333333333, 1.0)
+    >>> cdoit(4,2)
+    (2.0, 2.0)
+    """
+    cdef double a = x/y
+    cdef double b = x//y
+    return a, b
+
 def doit_inplace(x,y):
     """
     >>> doit_inplace(1,2)
